graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/DgsSchemaProvider.kt

```diff
@@ -445,31 +449,30 @@ class DgsSchemaProvider(
                 return@async method.kotlinFunction!!.callSuspend(dgsComponent, *args.toTypedArray())
             }
 
-            launch?.asCompletableFuture()
+            launch.asCompletableFuture()
         } else {
             ReflectionUtils.invokeMethod(method, dgsComponent, *args.toTypedArray())
         }
     }
 
     private fun getValueAsOptional(value: Any?, parameter: Parameter) =
-        if (value == null && parameter.type.isAssignableFrom(Optional::class.java)) {
-            Optional.empty<Any>()
-        } else if (parameter.type.isAssignableFrom(Optional::class.java)) {
-            Optional.of(value!!)
+        if (parameter.type.isAssignableFrom(Optional::class.java)) {
+            Optional.ofNullable(value)
```

Comment:
removing this unreachable branch here?

Reply:
I don't know that it was unreachable, just that it could be collapsed into a single case by switching to `Optional.ofNullable`, and thus reads a bit cleaner to me (and avoids `!!`).